src/components/TextField/TextField.tsx

```diff
@@ -129,394 +120,356 @@ export type TextFieldProps = NonMutuallyExclusiveProps &
+    },
+    [id, max, min, onChange, step, value],
+  );
+
+  const handleButtonRelease = useCallback(() => {
+    clearTimeout(buttonPressTimer.current);
+  }, [buttonPressTimer]);
+
+  const handleButtonPress = useCallback(
+    (onChange: Function) => {
+      const minInterval = 50;
+      const decrementBy = 10;
+      let interval = 200;
+
+      const onChangeInterval = () => {
+        if (interval > minInterval) interval -= decrementBy;
+        onChange();
+        buttonPressTimer.current = window.setTimeout(
+          onChangeInterval,
+          interval,
+        );
+      };
+
+      buttonPressTimer.current = window.setTimeout(onChangeInterval, interval);
+
+      addEventListener(document, 'mouseup', handleButtonRelease, {
+        once: true,
+      });
+    },
+    [handleButtonRelease],
+  );
+
+  const spinnerMarkup =
+    type === 'number' && !disabled && !readOnly ? (
+      <Spinner
+        onChange={handleNumberChange}
+        onMouseDown={handleButtonPress}
+        onMouseUp={handleButtonRelease}
+      />
     ) : null;
 
-    const characterCount = normalizedValue.length;
-    const characterCountLabel = intl.translate(
-      maxLength
-        ? 'Polaris.TextField.characterCountWithMaxLength'
-        : 'Polaris.TextField.characterCount',
-      {count: characterCount, limit: maxLength},
-    );
-
-    const characterCountClassName = classNames(
-      styles.CharacterCount,
-      multiline && styles.AlignFieldBottom,
-    );
+  const style = multiline && height ? {height} : null;
 
-    const characterCountText = !maxLength
-      ? characterCount
-      : `${characterCount}/${maxLength}`;
-
-    const characterCountMarkup = showCharacterCount ? (
-      <div
-        id={`${id}CharacterCounter`}
-        className={characterCountClassName}
-        aria-label={characterCountLabel}
-        aria-live={focus ? 'polite' : 'off'}
-        aria-atomic="true"
-      >
-        {characterCountText}
-      </div>
-    ) : null;
+  const handleExpandingResize = useCallback((height: number) => {
+    setHeight(height);
+  }, []);
 
-    const clearButtonMarkup =
-      clearButton && normalizedValue !== '' ? (
-        <button
-          testID="clearButton"
-          className={styles.ClearButton}
-          onClick={this.handleClearButtonPress}
-          disabled={disabled}
-        >
-          <VisuallyHidden>
-            {intl.translate('Polaris.Common.clear')}
-          </VisuallyHidden>
-          <Icon source={CircleCancelMinor} color="inkLightest" />
-        </button>
-      ) : null;
-
-    const spinnerMarkup =
-      type === 'number' && !disabled && !readOnly ? (
-        <Spinner
-          onChange={this.handleNumberChange}
-          onMouseDown={this.handleButtonPress}
-          onMouseUp={this.handleButtonRelease}
-        />
-      ) : null;
-
-    const style = multiline && height ? {height} : null;
-
-    const resizer = multiline ? (
+  const resizer =
+    multiline && mounted.current ? (
```

Comment:
I'm a little concerned about this, since in the hook when useEffect runs this will evaluate as false... but I could be being an airhead 😄